Support template variables metadata

lorenzo-cavazzi opened the pull request:
This PR implements the UI side to support template variables metadata, keeping backward compatibility with old template repositories.
The input elements bound to variables are now styled based on the underlying type, and enumerations are presented as dropdowns. Mind that there are differences between different browsers --  we can implement customization if needed.
The default value is pre-filled, unless a different value is otherwise defined (e.g. when sharing project creation links containing pre-filled data).

![Screenshot_20211119_145555](https://user-images.githubusercontent.com/43481553/142642185-e56289be-3950-4116-81c3-7c0a00b4bcc6.png)

**How to test**
This is a sample template repository where variables have metadata: https://dev.renku.ch/gitlab/lorenzo.cavazzi.tech/core-it-template-variable-test-project
If the reviewer wants to try some variations of that templates repo, the available metadata are: `description`, `default`, `type` and `enum` (only when "type" is also "enum").

**Further ideas**
* Should we adjust the components' sizes to occupy the whole width?
* Now that the page easily gets filled, it's easier to miss the pre-validation error we show close to the problematic field. Should we perhaps add a short error close to the "Create project" button as well? I myself missed the "Missing title" error a few times.

/deploy renku=1.0-next renku-graph=cli-v1 renku-core=2034-extended-template-variables
fix #1320

ciyer reviewed (changes requested):
This is nice and works very well! I agree that there should be some text next to the submit button that explains validation errors. 

I think we should use the `fa-undo` icon to for the "set value to default" button.

![image](https://user-images.githubusercontent.com/1196411/143031382-00a72f61-e492-42b4-b2c7-49f0be864441.png)

lorenzo-cavazzi commented:
I added the global error message and switched to the undo icon as suggested in the review.

Here are a couple of examples:

![Screenshot_20211123_170301](https://user-images.githubusercontent.com/43481553/143060142-f241ca35-7df6-4bbc-b19d-2d17c8dcb7a5.png)

![Screenshot_20211123_170336](https://user-images.githubusercontent.com/43481553/143060153-e755158b-b82e-4fd2-958f-94441778ad6f.png)
